diff --git a/dist/runtime/server/lib/oauth/github.mjs b/dist/runtime/server/lib/oauth/github.mjs
index 4f13049ec425087847b48e8e21a07beaca312997..fe7ae580fd6385ac2458b9ec4afe504f66bbca22 100644
--- a/dist/runtime/server/lib/oauth/github.mjs
+++ b/dist/runtime/server/lib/oauth/github.mjs
@@ -35,7 +35,7 @@ export function githubEventHandler({ config, onSuccess, onError }) {
       if (config.emailRequired && !config.scope.includes("user:email")) {
         config.scope.push("user:email");
       }
-      const redirectUrl = getRequestURL(event).href;
+      const redirectUrl = process.env.NUXT_OAUTH_GITHUB_REDIRECT_URI || getRequestURL(event).href;
       return sendRedirect(
         event,
         withQuery(config.authorizationURL, {
diff --git a/dist/runtime/server/lib/oauth/google.mjs b/dist/runtime/server/lib/oauth/google.mjs
index fa3cae0110948d53062ccd17a6e407d5636d6542..06a7766c719b34e5fe828a1d9feb3a9cfb83ae20 100644
--- a/dist/runtime/server/lib/oauth/google.mjs
+++ b/dist/runtime/server/lib/oauth/google.mjs
@@ -31,7 +31,7 @@ export function googleEventHandler({
         throw error;
       return onError(event, error);
     }
-    const redirectUrl = getRequestURL(event).href;
+    const redirectUrl = process.env.NUXT_OAUTH_GOOGLE_REDIRECT_URI || getRequestURL(event).href;
     if (!code) {
       config.scope = config.scope || ["email", "profile"];
       return sendRedirect(
